lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -3428,6 +3428,30 @@ OpFoldResult PrimMaxIntOp::fold(FoldAdaptor adaptor) {
       std::max(lhs.getValue().getSExtValue(), rhs.getValue().getSExtValue()));
 }
 
+//===----------------------------------------------------------------------===//
+// PrimNumToTensorScalarOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult PrimNumToTensorScalarOp::fold(FoldAdaptor adaptor) {
+  Attribute a = adaptor.getA();
+  auto resultTy = cast<BaseTensorType>(getType());
+  if (!a)
+    return {};
+  if (!resultTy.hasDtype() || !resultTy.hasSizes())
+    return {};
+
+  auto dty = resultTy.getDtype();
+  if (auto iattr = dyn_cast<IntegerAttr>(a)) {
+    a = IntegerAttr::get(dty, iattr.getInt());
+  } else if (auto fattr = dyn_cast<FloatAttr>(a)) {
+    a = FloatAttr::get(dty, fattr.getValueAsDouble());
+  }
+
+  auto builtin =
```

Comment:
It might be better to say "llvmTensorType" instead of "builtin" - - just to leave some breadcrumbs for anybody who might use this to learn to work on folders in the future.

Reply:
Seems reasonable, though I named it `mlirTensorType` as thats more accurate.